includes/admin/meta-boxes/class-wc-meta-box-product-data.php

```diff
@@ -189,7 +189,7 @@ public static function prepare_attributes( $data = false ) {
 			$attribute_names_max_key = max( array_keys( $attribute_names ) );
 
 			for ( $i = 0; $i <= $attribute_names_max_key; $i++ ) {
-				if ( empty( $attribute_names[ $i ] ) || ! isset( $attribute_values[ $i ] ) ) {
+				if ( empty( $attribute_names[ $i ] ) || ! isset( $attribute_values[ $i ] ) || is_string( $attribute_values[ $i ] ) && "" === trim( $attribute_values[ $i ] ) ) {
```

Comment:
Good catch. I thought the operator precedence worked differently than it actually does.